src/odemis/gui/model/__init__.py

```diff
@@ -1562,23 +1577,70 @@ def moveStageTo(self, pos):
+            move["y"] = max(stage_limits["y"][0], min(move["y"], stage_limits["y"][1]))
+            logging.info("Movements of the stage in y direction are limited by the stage limit, the movement is "
+                         "restricted to %s mm." % (move["y"] * 1e3))
 
         logging.debug("Requesting stage to move to %s m", move)
         f = self._stage.moveAbs(move)
         self._fstage_move = f
         f.add_done_callback(self._on_stage_move_done)
         return f
 
+    def _getStageLimitsXY(self):
+        """
+        Based on the physical stage limit and the area of the image used for imaging the stage limits are return in
+        a dict. (MD_POS_ACTIVE_RANGE defines the area which can be used for imaging)
+        If no stage limits are defined an empty dict is returned.
+
+        :return (dictionary): dict which contains the stage limits in x and y direction
+        """
+        stage_limits = {}
+        # Physical stage limits
+        if "x" in self._stage.axes and hasattr(self._stage.axes["x"], "range"):
+            stage_limits.update({"x": [self._stage.axes["x"].range[0], self._stage.axes["x"].range[1]]})
+        if "y" in self._stage.axes and hasattr(self._stage.axes["y"], "range"):
+            stage_limits.update({"y": [self._stage.axes["y"].range[0], self._stage.axes["y"].range[1]]})
+
+        # Area which can be used for imaging
+        if model.MD_POS_ACTIVE_RANGE in self._stage.getMetadata():
+            pos_active_range = self._stage.getMetadata()[model.MD_POS_ACTIVE_RANGE]
+            if "x" in pos_active_range and "x" in stage_limits:
+                stage_limits = self._updateStageLimits(stage_limits, {"x": pos_active_range["x"]})
+            if "y" in pos_active_range and "y" in stage_limits:
+                stage_limits = self._updateStageLimits(stage_limits, {"y": pos_active_range["y"]})
+
+        if not stage_limits:
+            logging.info("No stage limits defined")
+        return stage_limits
+
+    def _updateStageLimits(self, stage_limits, new_limits):
+        """
+        Updates the stage limits dictionary with the intersection of both the existing and new limits. So that the 
+        updated limits comply with both defined limits.
+
+        :param stage_limits (dict): Contains the limits for each axis of the stage which is limited.
+        :param new_limits (dict): Contains new limits for the stage for one or multiple axis
+        :return (dict): Contains the updated stage limits
+        """
+        for key in new_limits:
+            if key in stage_limits:
+                # Update the stage limits with the intersection of both the existing and new limits.
+                stage_limits.update({key: [max(stage_limits[key][0], new_limits[key][0]),
+                                           min(stage_limits[key][1], new_limits[key][1])]})
+            else:
+                stage_limits.update({key: [stage_limits[key][0], new_limits[key][1]]})
```

Comment:
I think this doesn't work if key is not in stage_limits. Shouldn't this be:
```
stage_limits.update({key: [new_limits[key][0], new_limits[key][1]]})
```
Which can then also be rewritten to:
```
stage_limits.update({key: new_limits[key]})
```

Reply:
That had to do with a needed conversion from tuple to list. But I fixed that.

Reply:
I also mean that your first argument is **stage**_limits[0] and your second argument is **new**_limits[1]